runtime/src/error.c

```diff
@@ -45,19 +49,52 @@
 #include <libunwind.h>
 
 #ifdef __linux__
+#include <dlfcn.h> // for dladdr
 // We create a pipe with addr2line and try to get a line number
 // Currently the precise line number works only on linux64
-static int chpl_unwind_getLineNum(char *process_name, void *addr){
-  char buf[128];
+static int chpl_unwind_getLineNum(void *addr){
+
+  // Compute the object containing the address
+  int rc;
+  Dl_info info;
+  intptr_t relativeAddr;
+  char process_name[256];
+  char buf[256];
```

Comment:
Given the usage, maybe `buf[100 + sizeof(process_name)]`?

Reply:
Is that valid in the C standard we target? (I would assume you know this better than I - but thought I'd check that you had thought about it)

Reply:
(Oh, I guess both of those are known at compile-time, so it's still a fixed size array, right?)

Reply:
No, it's not known at compile time, but this is a local (stack) variable, so it can have a size that is only determinable at during execution.

Reply:
Does the C standard we are using allow for variable-length arrays in a case like this? Like, could we size `buf` in a nested block like this:

```
const char* path = ...;
{
  char buf[strlen(path) + 100];
  ... do stuff with buf ...
}
```

I can never remember if such things are GNU extensions or C99 (and what C standard we are assuming).

Reply:
Arrays at local scopes have been allowed to have non-constant (i.e. execution-time) sizes since C99 at least.  There are bunches of these in the runtime already.

Reply:
(You wouldn't want to declare a gigabyte array this way of course, because you'd overflow the stack, but for little stuff like this it's fine.)

Reply:
Argh!  I'm sorry, I totally missed the point of your question.  Yes, `buf[100 + strlen(path)]` would be fine also.  Better, even.

Reply:
Since we need a big static sized buffer anyway for the readlink call - and since such dynamic arrays  size would still need to be checked - I went ahead with a strategy of only using 1 big buffer.